File: Data Structures & Algorithms/three-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
        nums.sort((a,b)=> a-b);
        const history = new Set();
        const output = [];
        const twoSum = (left, right, target) => {
            while(left < right) {
                const currentSum = nums[left] + nums[right];
                const serialized = `${-target}-${nums[left]}-${nums[right]}`;
                if(currentSum === target && !history.has(serialized)) {
                    output.push([-target, nums[left], nums[right]]);
                    history.add(serialized);
                    left++;
                    right--;
                } else if(currentSum < target) {
                    left++;
                } else {
                    right--;
                }
            }
        }
        for(let i = 0; i < nums.length - 2; i++){
            twoSum(i+1, nums.length -1, -nums[i]);
        }
        return output;
    }
}

File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        return self.merge(nums)

    def merge(self, nums: List[int]) -> List[int]:
        if len(nums) <= 1:
            return nums
        mid = len(nums) // 2
        left = self.merge(nums[:mid])
        right = self.merge(nums[mid:])
        output = []
        l_len = len(left)
        r_len = len(right)
        i = 0
        j = 0
        while i < l_len and j < r_len: 
            if left[i] <= right[j]:
                output.append(left[i])
                i += 1
            else:
                output.append(right[j])
                j += 1
        while i < l_len : 
            output.append(left[i])
            i += 1
        while j < r_len:
            output.append(right[j])
            j += 1
        return output
